Core/GDCore/IDE/Events/ArbitraryEventsWorker.cpp

```diff
@@ -165,6 +161,45 @@ void ReadOnlyArbitraryEventsWorker::StopAnyEventIteration() {
   shouldStopIteration = true;
 }
 
+ArbitraryEventsWorkerWithContext::~ArbitraryEventsWorkerWithContext() {}
+
+bool ArbitraryEventsWorkerWithContext::VisitEvent(gd::BaseEvent &event) {
+  // TODO Should CanHaveVariables be used instead?
+  // CanHaveVariables is less efficient but allows to add local variables to
+  // empty containers.
+  if (!event.HasVariables()) {
+    return ArbitraryEventsWorker::VisitEvent(event);
+  }
+  auto newProjectScopedContainers =
+      ProjectScopedContainers::MakeNewProjectScopedContainersWithLocalVariables(
+          *projectScopedContainers, event);
+
+  auto *parentProjectScopedContainers = projectScopedContainers;
+  projectScopedContainers = &newProjectScopedContainers;
+  bool shouldDelete = ArbitraryEventsWorker::VisitEvent(event);
+  projectScopedContainers = parentProjectScopedContainers;
```

Comment:
I wonder if a push/pop may be easier to read as the `projectScopedContainers` is a class member and not a parameter.

Reply:
It's a bit hard to read, but I think renaming the class member to "currentProjectScopedContainers" and a two comments "// Push a new 'scope' for variables" and "// Pop the 'scope' for variables" is enough.

("current" implies that you're going to change it and that it's just a pointer to whatever instance must be used currently, but it will change if needed.)